src/plugins/intel_gpu/src/plugin/transformations_pipeline.cpp

```diff
@@ -848,7 +848,16 @@ void TransformationsPipeline::apply(std::shared_ptr<ov::Model> func) {
         pass_config->disable<ov::pass::RoPEFusionIOSlicing>();
         pass_config->disable<ov::pass::RoPEShareCosSin>();
 
-        ov::pass::transformMLIR(func);
+        auto loweringContext = std::make_shared<ov::EvaluationContext>();
+        auto it = m_context->get_property().find(ov::intel_gpu::ocl_context.name());
+        if (it != m_context->get_property().end()) {
+            loweringContext->insert(ov::intel_gpu::ocl_context(it->second.as<ov::intel_gpu::gpu_handle_param>()));
+        }
+        void* ocl_device = m_context->get_engine().get_device()->get_handle();
+        if (ocl_device != nullptr) {
+            loweringContext->insert(ov::intel_gpu::va_device(ocl_device));
```

Comment:
I think that's not related to `VAAPI` at all, so this part looks confusing. If you need a device handle you can probably just extract it from context via clGetContextInfo

Reply:
> If you need a device handle you can probably just extract it from context via clGetContextInfo

Well, I've already [had exactly the same logic](https://github.com/dchigarev/openvino/blob/de413d45ab7088b4ae9cb8c80c1207904f17d5db/src/common/transformations/src/transformations/mlir/mlir_op.cpp#L347-L351) but wasn't sure enough if we can guarantee that there's always one device per context. If you would say that we can indeed guarantee that, then I can go back to the logic extracting device-id from `cl_context` in `MLIREvaluateGcGPU` constructor

Reply:
If context is created by GPU plugin, then we can expect that single device is used there. 
If context is created externally and shared with GPU plugin via RemoteAPI (which is not the main use case), then multiple devices may exist in single context, but request target device ID from the user. Refer to this part: https://github.com/openvinotoolkit/openvino/blob/master/src/inference/include/openvino/runtime/intel_gpu/ocl/ocl.hpp#L206

Currently we don't save this device ID, it's dropped once we extracted required cl::Device handle, but we can keep it if needed and then you'll be able to pass `intel_gpu::ocl_context` + `intel_gpu::ocl_context_device_id`. Though I think it's not mandatory for initial impl, we can have a restriction on single device for some time

Reply:
Okay, changed the logic to extract device ID from `cl_context` in `MLIREvaluateGcGPU` (f0ecd164c12d8ae6f0c0dabf2e9f27baef27d22f)